sdk/core/Azure.Core/src/RequestUriBuilder.cs

```diff
@@ -1,355 +1,14 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
-using System.Diagnostics;
-using System.IO;
-using System.Text;
+using System.ServiceModel.Rest.Shared.Core;
 
 namespace Azure.Core
 {
     /// <summary>
     /// Provides a custom builder for Uniform Resource Identifiers (URIs) and modifies URIs for the <see cref="System.Uri" /> class.
     /// </summary>
-    public class RequestUriBuilder
+    public class RequestUriBuilder : RequestUri
```

Comment:
This pattern with four types and one in shared source with copied code and an adapter makes me very, very sad.  Let's definitely clean this up, I am interested in looking at what we could do by moving RawRequestUriBuilder functionality into RequestUriBuilder, and possibly all just up to one top level type in the System.ServiceModel.Rest library.  Shared source makes this hard to do today.